core.md

```diff
@@ -89,10 +113,35 @@ struct would:
     | y: float64 |      |
     +============+    --+
 
-Structure definitions can also reference other structure
-definitions:
+##### Fixed-sized indexed structs
+
+```js
+const LineType = new StructType(PointType, 2);
+```
+
+This defines a new type definition `LineType` that consists of two indexed
+elements, each an instance of `PointType`. These will be laid out in memory consecutively, just as a C struct would:
+
+    +===============+    --+ LineType
+    | 0: x: float64 |      | --+ PointType
+    |    y: float64 |      | --+
+    | 1: x: float64 |      |
+    |    y: float64 |      |
+    +===============+    --+
+
+An equivalent definition to this, that'd become unwieldy for large `length`s, would be:
+
+```js
+const LineType = new StructType({0: PointType, 1: PointType});
+```
```

Comment:
I changed the example to include a `length` field. I think it'd be a shame not to really make the two definitions equivalent, and really don't see too much of a reason not to.